packages/repository/src/repositories/constraint-utils.ts

```diff
@@ -71,13 +74,7 @@ export function constrainDataObject<T extends Entity>(
  */
 export function constrainDataObjects<T extends Entity>(
   originalData: DataObject<T>[],
-  constraint: Partial<T>,
+  constraint: DataObject<T>,
 ): DataObject<T>[] {
-  const constrainedData = cloneDeep(originalData);
-  for (let obj of constrainedData) {
-    for (let prop in constraint) {
-      (obj as AnyObject)[prop] = constraint[prop];
-    }
-  }
-  return constrainedData;
+  return originalData.map(obj => constrainDataObject(obj, constraint));
```

Comment:
This is slightly changing the behavior when the input data is trying to modify the constrained properties. Before my change, we would silently replace values provided by the user with values enforced by the constraint. With my change in place, we reject requests trying to modify constrained properties.

AFAICT, the function `constrainDataObjects` (notice the plural name!) is not used anywhere in loopback-next codebase and I would find it surprising if anybody was using it directly from 3rd party projects. I consider this change as a backwards-compatible bug fix (not a breaking change).